src/om_bootstrap/modal.cljs

```diff
@@ -1,104 +1,54 @@
-;;
-;; https://gist.github.com/insin/8449696
-#_
-(defcomponentk modal
-  "Component to render a Bootstrap modal to the DOM."
-  [[:data animation? on-request-hide {keyboard? true} :as modal] :- Modal
-   owner state]
-  (:mixins fade-mixin set-listener-mixin)
-  (init-state [_] {:keyup (chan)})
-  (will-mount [_]
-              (let [c (:keyup @state)]
-                (go-loop []
-                  (let [item (a/<! c)]
-                    (if (and keyboard? (.-keycode item))
-                      (on-request-hide)))
-                  (recur))))
-  (did-mount [_] (.set-listener owner "keyup" (:keyup @state)))
+(defcomponentk modal*
+  "Component that renders a Modal. Manages it's own toggle state"
+  [owner state]
+  (init-state [_]
+    {:visible? (-> (om/get-props owner)
+                   :opts
+                   :visible?)})
   (render [_]
-          (d/div {} (d/p {} "Placeholder."))
-          #_(let [style {:display "block"}
-                  klasses {:modal true
-                           :fade animation?
-                           :in (or (not animation?)
-                                   (not (.querySelectorAll js/document)))}
-                  modal-elem (d/div {:title nil
-                                     :tab-index "-1"
-                                     :role "dialog"
-                                     :class (d/class-set klasses)
-                                     :ref "modal"}
-                                    (d/div {:class "modal-dialog"}
-                                           (d/div {:class "modal-content"}
-                                                  (when (:title modal)
-                                                    (render-header modal))
-                                                  (:children modal))))]
-              (if (:backdrop? modal)
-                (render-backdrop modal modal-elem)
-                modal-elem))))
+    (let [{:keys [opts children]} (om/get-props owner)
+          [bs props] (t/separate Modal opts {:bs-class "modal"})
+          visible? (om/get-state owner [:visible?])
+          classes {:modal true
+                   :fade true
+                   :in visible?}]
+      (d/div (u/merge-props props
+                            {:class (d/class-set classes)})
+             (d/div {:class "modal-dialog"}
+                    (d/div {:class "modal-content"}
+                           (d/div {:class "modal-header"}
+                                  (when (:close-button? bs)
+                                    (d/button {:type         "button"
+                                               :class        "close"
+                                               :aria-hidden  true
+                                               :on-click (fn [_] (hide! owner))}
+                                              "×"))
+                                  (u/clone-with-props (:header bs) {:class "modal-title"}))
```

Comment:
Should we leave the actual header element to the user or should we wrap it in a ```h4``` by default?